pgx/src/datum/numeric_support/cmp.rs

```diff
@@ -0,0 +1,126 @@
+use std::cmp::Ordering;
+
+use crate::{direct_function_call, pg_sys, AnyNumeric, Numeric};
+
+impl PartialEq<AnyNumeric> for AnyNumeric {
+    #[inline]
+    fn eq(&self, other: &AnyNumeric) -> bool {
+        unsafe {
+            direct_function_call(pg_sys::numeric_eq, vec![self.as_datum(), other.as_datum()])
+                .unwrap()
+        }
+    }
+
+    #[inline]
+    fn ne(&self, other: &AnyNumeric) -> bool {
+        unsafe {
+            direct_function_call(pg_sys::numeric_ne, vec![self.as_datum(), other.as_datum()])
+                .unwrap()
+        }
+    }
+}
+
+impl Eq for AnyNumeric {}
+
+impl PartialOrd for AnyNumeric {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
```

Comment:
This should handle the case of one value being NaN

Reply:
Postgres provides total ordering for NUMERIC, so we can blindly accept what `pg_sys::numeric:cmp()` tells us.  I've cleaned it up in eeffdfb9f3c294914ed58608c181fae096d81805.